src/Collections-Atomic/WaitfreeQueue.class.st

```diff
@@ -295,3 +365,13 @@ WaitfreeQueue >> size [
 
 	^ count
 ]
+
+{ #category : #initialization }
+WaitfreeQueue >> withAll: aCollection [
+	"Initialise the receiver with the supplied items"
+
+	dummy next: aCollection first.
+	1 to: aCollection size - 1 do: [ :i |
+		(aCollection at: i) next: (aCollection at: i+1) ].
+	tail := aCollection last.
```

Comment:
And could you explain why it is not implemented using #nextPut:?
I see that comment points to initialization of queue. Is it difficult to have it as a common #nextPutAll: method? 
For initialization only I would propose name #initializeWithAll:.

Reply:
> And could you explain why it is not implemented using #nextPut:?
> I see that comment points to initialization of queue. Is it difficult to have it as a common #nextPutAll: method?

For testing I want to be able to set up the items in a pre-determined manner, e.g. be able to add zombie items.  That wouldn't be possible with `#nextPutAll:`.


> For initialization only I would propose name #initializeWithAll:.

Good suggestion, I'll change the name.